### DISCUSSION TITLE
Licensing requirement clarification

### DISCUSSION THREAD

**Discussion by thegcat** [pretix/pretix#2043] [Q&A] [answered]
We, the [Fachschaft Informatik der TU Dortmund](https://fsinfo.cs.tu-dortmund.de), are currently running a Pretix installation that is used by us but also other entities, such as the [Konferenz der Informatikfachschaften (KIF)](https://kif.rocks) and the [Konferenz der Mathematikfachschaften (KoMa)](https://die-koma.org). As such I would say we fall in the "we run Pretix for others" regarding the license and more specifically [this FAQ entry](https://docs.pretix.eu/en/latest/license/faq.html#what-do-i-need-to-do-if-i-modify-pretix).

We are currently using a plugin on that instance, to which the above FAQ entry states:

> If you run it for others, for example as part of a Software-as-a-Service offering or a managed hosting service you must make the source code including all your modifications and all installed plugins available under the same license as pretix to every visitor of your site. You need to do so in a prominent place such as a link at the bottom of the site.

For the moment we have used the custom footer text and link from in the global settings to link to said plugin, see [one of the custom domains for this Pretix instance](https://tix.kif.rocks), which from my understanding satisfies the license condition.

How would we go about mentioning and linking to the plugins used in case we have more than one plugin? Does Pretix offer some sort of publicly reachable list of installed plugins one could link to from said footer? Would a link "this Pretix uses several plugins, click here for more info" linking to an external site listing the installed plugins satisfy the requirement?

Thank you for your help in this matter.

**Comment by raphaelm** [comment, marked as answer]
> We, the Fachschaft Informatik der TU Dortmund, are currently running a Pretix installation that is used by us but also other entities, such as the Konferenz der Informatikfachschaften (KIF) and the Konferenz der Mathematikfachschaften (KoMa). As such I would say we fall in the "we run Pretix for others" regarding the license and more specifically this FAQ entry.

Yes, I'd agree, you'd technically fall under that definition and need to comply with AGPLv3.

> How would we go about mentioning and linking to the plugins used in case we have more than one plugin? Does Pretix offer some sort of publicly reachable list of installed plugins one could link to from said footer?

Yes! The current master branch and upcoming release include a "license check & settings" section in the backend which includes a big markdown text box where you can link the source code of all plugin. The content will show up at ``/agpl_source`` on all domains and will be liked in the footer automatically :)

**Reply by thegcat** [reply]
> Yes! The current master branch and upcoming release include a "license check & settings" section in the backend which includes a big markdown text box where you can link the source code of all plugin. The content will show up at `/agpl_source` on all domains and will be liked in the footer automatically :)

Sounds great!

Would it also be possible to just have the list generated? Or maybe just a button that would pre-populate the text box with the current list of non-core plugins and their URLs?

**Reply by raphaelm** [reply]
Wait and see :) It does that, where it can

**Reply by thegcat** [reply]
Looks great, thanks.

if it is of any importance: The "check your license" banner was shown directly after I logged in though I was not in Admin mode, and I was able to make changes to that page without being in Admin mode.

**Reply by raphaelm** [reply]
Yes, you need to *be* a superuser, but you do not need to be in admin mode, I think that was intentional.

**Reply by thegcat** [reply]
OK. The menu entry is only rendered when in Admin mode though, so maybe this should be adapted.
